google-startup-scripts/usr/share/google/onboot

```diff
@@ -112,6 +112,41 @@ function print_ssh_key_fingerprints() {
   return 0
 }
 
+function check_for_connection() {
+  local count=0
+  local return_code=0
+  local url="http://metadata/computeMetadata/v1/instance/?recursive=True"
+
+  log "Checking for metadata server connection."
+  while true; do
+    ((count++))
+    curl -f ${url} -H "Metadata-Flavor: Google" 2>/dev/null
+    return_code=$?
+    case $return_code in
+      # No error.  Connection is active.
+      0) break;;
+      # Failed to resolve host or connect to host.  Retry indefinitely.
+      6|7) sleep 1.0
+        log "Waiting for metadata server, attempt ${count}"
+        # After 7 minutes, add a console message denoting a probable network
+        # issue. On systems using dhclient there is an attempt to obtain an IP
+        # for 60 seconds followed by a 5 minute wait period. After 7 minutes,
+        # this cycle will have run through twice. After this period of time, it
+        # is not known when a DHCP lease might be obtained and the network
+        # interface fully operational.
+        if ((count >= 421)); then
```

Comment:
I see where 421 came from, but can you make it obvious in the code? E.g. 7*60+1 with the comment making it clear this is in seconds.